### PR TITLE
Minor memory + perf export optimizations

### PR DESCRIPTION
**Pull request by dralley**
[noissue]

### CODE REVIEW

File: pulpcore/app/importexport.py [hunk 1]
```diff
@@ -48,15 +48,29 @@ def _write_export(the_tarfile, resource, dest_dir=None):
...
+            batch = []
+            needs_comma = False
+            for item in resource.queryset.iterator(chunk_size=EXPORT_BATCH_SIZE):
+                batch.append(item)
+                if needs_comma:
                     # Write "," if not last loop
                     temp_file.write(", ")
+                    needs_comma = False
+
+                if len(batch) >= EXPORT_BATCH_SIZE:
```
Comment (dralley) [inline pulpcore/app/importexport.py:66]: Is there a more elegant way to express all of this?

**Comment by dralley** [comment]
Saves about 5-8% on time and memory.

File: pulpcore/app/importexport.py (hunk 1 above)
Comment (mdellweg) [reply pulpcore/app/importexport.py:66]: This is what i constantly come up with when writing data migrations.

File: pulpcore/app/tasks/export.py [hunk 2]
```diff
@@ -504,16 +509,16 @@ def _do_export(pulp_exporter, tar, the_export):
...
-        artifacts.update(vers_artifacts)
+            vers_artifacts = version.artifacts
+        artifact_pks.update(vers_artifacts.values_list("pk", flat=True))
 
-    # export plugin-version-info
-    export_versions(the_export, plugin_version_info)
     # Export the top-level entities (artifacts and repositories)
     # Note: we've already handled "what about incrementals" when building the 'artifacts' list
-    export_artifacts(the_export, list(artifacts))
+    export_artifacts(the_export, Artifact.objects.filter(pk__in=artifact_pks))
```
Comment (mdellweg) [inline pulpcore/app/tasks/export.py:519]: Is this qs the new memory hog now?

**Review by mdellweg — APPROVED** [review]
This looks good to me. Hope we have good test coverage here.

**Review by ipanova — APPROVED** [review]

File: pulpcore/app/tasks/export.py (hunk 2 above)
Comment (dralley) [reply pulpcore/app/tasks/export.py:519]: I don't believe so.  It didn't seem to move the needle much when I executed it in a shell.  It's lazily evaluated in any case so it's not using any memory here.

**Comment by dralley** [comment]
>Hope we have good test coverage here.

I don't know that it's amazingly thorough but I'm heartened by the fact that it has caught things in the past.

**Pull request merged**